src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -333,47 +351,48 @@ IsCitusTableTypeCacheEntry(CitusTableCacheEntry *tableEntry, CitusTableType tabl
  * the given table type group. For definition of table types, see CitusTableType.
  */
 static bool
-IsCitusTableTypeInternal(CitusTableCacheEntry *tableEntry, CitusTableType tableType)
+IsCitusTableTypeInternal(char partitionMethod, char replicationModel,
```

Comment:
umm, here I am a bit skeptical about giving `partitionMethod` and `replicationModel`. The point of passing `cacheEntry` here was that it might change what a `reference` table is identified by, but that information will be in the cache entry. Currently it is `partitionMethod` and `replicationModel` but this might change. 

I wonder if we should prefer the `cacheEntry` approach with manually filtering some fields in `IsCitusTableViaCatalogInternal`. In that case if something changes in the future we won't need to change `IsCitusTableTypeInternal` and we will only need to update those fields of cache entry in the right places.